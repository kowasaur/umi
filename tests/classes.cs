class Thing {
    public static string str = "";
    public int num = 0;

    public Thing() {}
}
