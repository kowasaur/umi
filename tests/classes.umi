class Person {
    string name
    int32 age

    Person(string name, int32 a) {
        this.name = name
        age = a
        print("Making " + name + " caused a side effect")
    }
}

void main() {
    Person p = Person("Barry", 63)
    print(p.name)
    print(p.age)
}
