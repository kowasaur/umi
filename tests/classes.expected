Making Larry caused a side effect
Larry
63
Hello I am Barry, 63
78
10
42
This is a message
pog
I don't know what to write
This is a message
pog
I don't know what to write

3
Thing string
